hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestFileSystemInitialization.java

```diff
@@ -77,6 +77,22 @@ public void testNewInstanceFailure() throws Throwable {
         .isEqualTo(1);
   }
 
+  @Test
+  public void testInvalidLocalFileSystem() throws Exception {
+    Configuration conf = new Configuration();
+    conf.set("fs.file.impl", "org.apache.hadoop.fs.RawLocalFileSystem");
+    try {
```

Comment:
use LambdaTestUtils.intercept(); return the actual FileSystem.toString() in the error, e.g
```java
intercept(IllegalArgument.class, "should extend", () -> 
  "got filesystem " + FileSystem.getLocal(conf));
```